lib/devise/failure_app.rb

```diff
@@ -252,5 +246,17 @@ def is_flashing_format?
     def request_format
       @request_format ||= request.format.try(:ref)
     end
+
+    def relative_url_root
+      @relative_url_root ||= begin
+        config = Rails.application.config
+
+        config.try(:relative_url_root) || config.try(:action_controller).try(:relative_url_root)
```

Comment:
There is any scenario where the `config.action_controller` call would fail for it to require the `try` approach on it?